### PR TITLE
Fixes #1846

### LINKED ISSUE #1846: Bidirectional Transport Lumen <=> Mucosa: wrong definition and ProcessRate parameter not created
1) Transport definition is wrong. There are no target conditions and source conditions do not make sense.
grafik
Thus the transport is not created in simulations at all.
grafik

Instead, source/target conditions must be defined ~~either as for the `Influx Lumen==>Mucosa`~~ or as for the `Efflux Mucosa ==> Lumen` (EDIT: just checked where the transporter is located: is in Mucosa. So the source must be mucosa and target lumen)
2) ProcessRate parameter is not created as per Default. This would lead to the wrong calculation of fraction absorbed (s. 1845)
grafik

### PR DESCRIPTION
**Pull request by msevestre**

### CODE REVIEW

File: src/Db/Diff/PKSimDB_diff.txt [hunk 1]
```diff
@@ -1,57 +1,5 @@
...
-INSERT INTO tab_rate_generic_parameters(calculation_method,formula_rate,path_id,parameter_name,alias) VALUES('Lumen_PKSim','PARAM_OralMassAbsorbed_RHS_uje',2126,'Sum of active process rates lumen to mucosa','ActiveRates_para_OUT');
-UPDATE tab_rate_generic_parameters SET alias='ActiveRates_para_IN' WHERE calculation_method='Lumen_PKSim' AND formula_rate='PARAM_OralMassAbsorbed_RHS_uje' AND path_id=2126 AND parameter_name='Sum of active process rates mucosa to lumen';
-INSERT INTO tab_rate_generic_parameters(calculation_method,formula_rate,path_id,parameter_name,alias) VALUES('Lumen_PKSim','PARAM_OralMassAbsorbed_RHS_uje',2131,'Sum of active process rates lumen to mucosa','ActiveRates_trans_OUT');
-UPDATE tab_rate_generic_parameters SET alias='ActiveRates_trans_IN' WHERE calculation_method='Lumen_PKSim' AND formula_rate='PARAM_OralMassAbsorbed_RHS_uje' AND path_id=2131 AND parameter_name='Sum of active process rates mucosa to lumen';
+INSERT INTO tab_process_descriptor_conditions(process,tag,tag_type,should_have) VALUES('ActiveBiDirectionalSpecificLumenMucosaIntracellular_MM','Feces','TARGET',0);
```
Comment (msevestre) [inline src/Db/Diff/PKSimDB_diff.txt:1]: @Yuri05 Forgotten indeed. In that case, the algorithm is BS

File: src/Db/Dump/PKSimDB_dump.txt [hunk 2]
```diff
@@ -80950,7 +80950,7 @@ INSERT INTO tab_processes VALUES('ActiveBiDirectionalSpecificInterstitialIntrace
 INSERT INTO tab_processes VALUES('ActiveBiDirectionalSpecificBloodCellsPlasma','PROCESS',1,'INDIVIDUAL_ACTIVE_PROCESS','Undefined','BiDirectional','TRANSPORT',0);
 INSERT INTO tab_processes VALUES('ActiveBiDirectionalSpecificBloodCellsPlasma_MM','PROCESS',1,'SIMULATION_ACTIVE_PROCESS','MM','BiDirectional','TRANSPORT',0);
 INSERT INTO tab_processes VALUES('ActiveBiDirectionalSpecificLumenMucosaIntracellular','PROCESS',1,'INDIVIDUAL_ACTIVE_PROCESS','Undefined','BiDirectional','TRANSPORT',0);
-INSERT INTO tab_processes VALUES('ActiveBiDirectionalSpecificLumenMucosaIntracellular_MM','PROCESS',1,'SIMULATION_ACTIVE_PROCESS','MM','BiDirectional','TRANSPORT',0);
+INSERT INTO tab_processes VALUES('ActiveBiDirectionalSpecificLumenMucosaIntracellular_MM','PROCESS',1,'SIMULATION_ACTIVE_PROCESS','MM','BiDirectional','TRANSPORT',1);
```
Comment (msevestre) [inline src/Db/Dump/PKSimDB_dump.txt:80953]: Also set the parameter to 1... Why is this required?

File: src/Db/Diff/PKSimDB_diff.txt [hunk 3]
```diff
@@ -1,57 +1,8 @@
...
+INSERT INTO tab_process_descriptor_conditions(process,tag,tag_type,should_have) VALUES('ActiveBiDirectionalSpecificLumenMucosaIntracellular_MM','Intracellular','SOURCE',1);
+INSERT INTO tab_process_descriptor_conditions(process,tag,tag_type,should_have) VALUES('ActiveBiDirectionalSpecificLumenMucosaIntracellular_MM','Lumen','TARGET',1);
+INSERT INTO tab_process_descriptor_conditions(process,tag,tag_type,should_have) VALUES('ActiveBiDirectionalSpecificLumenMucosaIntracellular_MM','Mucosa','SOURCE',1);
+DELETE FROM tab_process_descriptor_conditions WHERE process='ActiveBiDirectionalSpecificPlasmaInterstitial_MM' AND tag='VenousBlood' AND tag_type='SOURCE';
```
Comment (msevestre) [inline src/Db/Diff/PKSimDB_diff.txt:5]: Remove the 3 VenousBlood tags...no instertitial there anyways

File: src/PKSim.UI/Views/Simulations/SimulationExpressionsView.cs [hunk 4]
```diff
@@ -23,6 +23,7 @@ public override void InitializeResources()
       {
          base.InitializeResources();
          layoutGroupMoleculeParameters.Text = PKSimConstants.UI.Properties;
+         layoutGroupMoleculeParameters.ExpandButtonVisible = true;
```
Comment (msevestre) [inline src/PKSim.UI/Views/Simulations/SimulationExpressionsView.cs:26]: and this issue also #1843

File: src/Db/Dump/PKSimDB_dump.txt (hunk 2 above)
Comment (Yuri05) [reply src/Db/Dump/PKSimDB_dump.txt:80953]: Because we need ProcessRate parameters of all transports (passive and active) between lumen and mucosa to calculate the fraction absorbed.
(dMassAbsorbed/dt is calculated as sum of all process rates)

File: src/Db/Dump/PKSimDB_dump.txt [hunk 5]
```diff
@@ -81513,15 +81513,16 @@ INSERT INTO tab_process_descriptor_conditions VALUES('ActiveInfluxSpecificPlasma
 INSERT INTO tab_process_descriptor_conditions VALUES('ActiveInfluxSpecificPlasmaToBloodCells_MM','Plasma','SOURCE',1);
 INSERT INTO tab_process_descriptor_conditions VALUES('ActiveEffluxSpecificInterstitialToPlasma_MM','VenousBlood','TARGET',0);
 INSERT INTO tab_process_descriptor_conditions VALUES('ActiveEffluxSpecificInterstitialToPlasma_Hill','VenousBlood','TARGET',0);
-INSERT INTO tab_process_descriptor_conditions VALUES('ActiveInfluxSpecificPlasmaToInterstitial_Hill','VenousBlood','SOURCE',0);
-INSERT INTO tab_process_descriptor_conditions VALUES('ActiveInfluxSpecificPlasmaToInterstitial_MM','VenousBlood','SOURCE',0);
```
Comment (Yuri05) [inline src/Db/Dump/PKSimDB_dump.txt:81517]: ok, this is wrong and must be reactivated again. We need the `NOT Venousblood `condition for all active transports Plasma<=>Interstitial because we have nbhoods between Venous/Plasma and TissueOrgan/Interstitial in the protein model (and we only want active transports between plasma and interstitial of the same organ)

File: src/Db/Dump/PKSimDB_dump.txt [hunk 6]
```diff
@@ -81513,15 +81513,16 @@ INSERT INTO tab_process_descriptor_conditions VALUES('ActiveInfluxSpecificPlasma
...
 INSERT INTO tab_process_descriptor_conditions VALUES('ActiveEffluxSpecificInterstitialToPlasma_MM','VenousBlood','TARGET',0);
 INSERT INTO tab_process_descriptor_conditions VALUES('ActiveEffluxSpecificInterstitialToPlasma_Hill','VenousBlood','TARGET',0);
-INSERT INTO tab_process_descriptor_conditions VALUES('ActiveInfluxSpecificPlasmaToInterstitial_Hill','VenousBlood','SOURCE',0);
-INSERT INTO tab_process_descriptor_conditions VALUES('ActiveInfluxSpecificPlasmaToInterstitial_MM','VenousBlood','SOURCE',0);
 INSERT INTO tab_process_descriptor_conditions VALUES('ActiveBiDirectionalSpecificInterstitialIntracellular_MM','Interstitial','SOURCE',1);
 INSERT INTO tab_process_descriptor_conditions VALUES('ActiveBiDirectionalSpecificInterstitialIntracellular_MM','Intracellular','TARGET',1);
 INSERT INTO tab_process_descriptor_conditions VALUES('ActiveBiDirectionalSpecificPlasmaInterstitial_MM','Interstitial','TARGET',1);
 INSERT INTO tab_process_descriptor_conditions VALUES('ActiveBiDirectionalSpecificPlasmaInterstitial_MM','Plasma','SOURCE',1);
-INSERT INTO tab_process_descriptor_conditions VALUES('ActiveBiDirectionalSpecificPlasmaInterstitial_MM','VenousBlood','SOURCE',0);
```
Comment (Yuri05) [inline src/Db/Dump/PKSimDB_dump.txt:81522]: same

**Review by Yuri05 — CHANGES_REQUESTED** [review]
s. comments

File: src/Db/Diff/PKSimDB_diff.txt [hunk 7]
```diff
@@ -1,57 +1,9 @@
...
+ALTER TABLE tab_process_descriptor_conditions ADD COLUMN comment;
+INSERT INTO tab_process_descriptor_conditions(process,tag,tag_type,should_have,comment) VALUES('ActiveBiDirectionalSpecificLumenMucosaIntracellular_MM','Feces','TARGET',0,NULL);
+INSERT INTO tab_process_descriptor_conditions(process,tag,tag_type,should_have,comment) VALUES('ActiveBiDirectionalSpecificLumenMucosaIntracellular_MM','Intracellular','SOURCE',1,NULL);
+INSERT INTO tab_process_descriptor_conditions(process,tag,tag_type,should_have,comment) VALUES('ActiveBiDirectionalSpecificLumenMucosaIntracellular_MM','Lumen','TARGET',1,NULL);
+INSERT INTO tab_process_descriptor_conditions(process,tag,tag_type,should_have,comment) VALUES('ActiveBiDirectionalSpecificLumenMucosaIntracellular_MM','Mucosa','SOURCE',1,NULL);
+UPDATE tab_process_descriptor_conditions SET comment='we have nbhoods between Venous/Plasma and TissueOrgan/Interstitial in the protein model (and we only want active transports between plasma and interstitial of the same organ)' WHERE process='ActiveBiDirectionalSpecificPlasmaInterstitial_MM' AND tag='VenousBlood' AND tag_type='SOURCE';
```
Comment (msevestre) [inline src/Db/Diff/PKSimDB_diff.txt:6]: @Yuri05 Ok added a commment column so that we can remember why this is like this in the future

Comment (Yuri05) [reply src/Db/Diff/PKSimDB_diff.txt:6]: cool.
btw, this PR is one of the rare examples, where DB-changes are much easier to see in **PKSimDB_diff** than in **PKSimDB_dump**.

Comment (msevestre) [reply src/Db/Diff/PKSimDB_diff.txt:6]: yep... That's why we have diff and dump :) This was an amazing addition

**Comment by msevestre** [comment]
@Yuri05 Can we merge?

**Review by Yuri05 — APPROVED** [review]
